core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -538,10 +538,37 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     }
   }
 
+  test("Killing tasks") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+
+    SparkContextSuite.isTaskStarted = false
+    SparkContextSuite.taskKilled = false
+
```

Comment:
Can you add a comment about what this test is doing?  e.g.,

Launches one task that will run forever.  Once the SparkListener detects the task has started, kill and re-schedule it.  The second run of the task will complete immediately.  If this test times out, then the first version of the task wasn't killed successfully.